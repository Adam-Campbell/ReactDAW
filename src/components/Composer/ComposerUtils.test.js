import { 
    createCopiedSectionsDataStructure,
    findEarliestStartTime,
    createGridLinesArray,
    createSectionRectsArray,
    createSectionObject
} from './ComposerUtils';

const channelsState = [
    {
      id: '1146587645959972',
      name: 'Channel 1',
      color: '#00bff3',
      instrumentId: '4670010528826357',
      effectIds: [],
      sectionIds: [
        '3873914619840634',
        '5586134390473357',
        '4928704574589138'
      ]
    },
    {
      id: '2532017111888654',
      name: 'Channel 2',
      color: '#f4ab54',
      instrumentId: '7909979211308588',
      effectIds: [],
      sectionIds: [
        '5208036813028087',
        '6955590810822236',
        '9199138957540981'
      ]
    },
    {
      id: '8988686808971398',
      name: 'Channel 3',
      color: '#d86597',
      instrumentId: '9916153873524561',
      effectIds: [],
      sectionIds: []
    }
];
const sectionsState = {
    '5208036813028087': {
      id: '5208036813028087',
      channelId: '2532017111888654',
      notes: [],
      start: '0:0:0',
      numberOfBars: 4
    },
    '6955590810822236': {
      id: '6955590810822236',
      channelId: '2532017111888654',
      notes: [],
      start: '4:0:0',
      numberOfBars: 4
    },
    '3873914619840634': {
      id: '3873914619840634',
      channelId: '1146587645959972',
      notes: [],
      start: '4:0:0',
      numberOfBars: 1
    },
    '5586134390473357': {
      id: '5586134390473357',
      channelId: '1146587645959972',
      notes: [],
      start: '6:0:0',
      numberOfBars: 1
    },
    '4928704574589138': {
      id: '4928704574589138',
      channelId: '1146587645959972',
      notes: [],
      start: '8:0:0',
      numberOfBars: 2
    },
    '9199138957540981': {
      id: '9199138957540981',
      channelId: '2532017111888654',
      notes: [],
      start: '10:0:0',
      numberOfBars: 2
    }
};

describe('createCopiedSectionsDataStructure', () => {
    test('correctly creates and returns the desired data structure', () => {
        const selectionState = [
            '5208036813028087', 
            '3873914619840634',
            '5586134390473357',
            '9199138957540981',
        ];
        const expectedResult = {
            lowestIndex: 0,
            sectionObjects: [
                {
                    id: '5208036813028087',
                    channelIndex: 1,
                    notes: [],
                    start: '0:0:0',
                    numberOfBars: 4
                },
                {
                    id: '3873914619840634',
                    channelIndex: 0,
                    notes: [],
                    start: '4:0:0',
                    numberOfBars: 1
                },
                {
                    id: '5586134390473357',
                    channelIndex: 0,
                    notes: [],
                    start: '6:0:0',
                    numberOfBars: 1
                },
                {
                    id: '9199138957540981',
                    channelIndex: 1,
                    notes: [],
                    start: '10:0:0',
                    numberOfBars: 2
                }
            ]
        };
        const result = createCopiedSectionsDataStructure({
            currentSelectionState: selectionState,
            allSections: sectionsState,
            allChannels: channelsState
        });
        expect(result).toEqual(expectedResult);
    });
});


describe('findEarliestStartTime', () => {
    test('correctly identifies the earliest start time from the provided section objects', () => {
        const sectionObjects = [
            {
                id: '5586134390473357',
                channelIndex: 0,
                notes: [],
                start: '6:0:0',
                numberOfBars: 1
            },
            {
                id: '3873914619840634',
                channelIndex: 0,
                notes: [],
                start: '4:0:0',
                numberOfBars: 1
            },
            {
                id: '5208036813028087',
                channelIndex: 1,
                notes: [],
                start: '1:0:0',
                numberOfBars: 4
            },
            {
                id: '9199138957540981',
                channelIndex: 1,
                notes: [],
                start: '10:0:0',
                numberOfBars: 2
            }
        ];
        const result = findEarliestStartTime(sectionObjects);
        expect(result).toBe('1:0:0');
    });
    test('returns null if called with an empty array', () => {
        const result = findEarliestStartTime([]);
        expect(result).toBe(null);
    });
});

describe('createGridLinesArray', () => {
    test('correctly creates and returns the array', () => {
        const result1 = createGridLinesArray({
            canvasHeight: 320,
            canvasWidth: 9600,
            channelsArrayLength: 0 
        });
        expect(result1).toMatchSnapshot();
        const result2 = createGridLinesArray({
            canvasHeight: 530,
            canvasWidth: 12000,
            channelsArrayLength: 7
        });
        expect(result2).toMatchSnapshot();
    });
});

describe('createSectionRectsArray', () => {
    test('correctly creates and returns the array', () => {
        const result = createSectionRectsArray({
            allChannels: channelsState,
            allSections: sectionsState
        });
        expect(result).toMatchSnapshot();
    });
});

describe('createSectionObject', () => {
    test('correctly creates and returns the section object if the inputs are valid', () => {
        const result = createSectionObject({
            x: 96,
            y: 220,
            allChannels: channelsState,
            numberOfBars: 2
        });
        expect(result.channelId).toBe('8988686808971398');
        expect(result.numberOfBars).toBe(2);
        expect(result.start).toBe('2:0:0');
    });
    test('returns null if the inputs are not valid', () => {
        const result = createSectionObject({
            x: 48,
            y: 350,
            allChannels: channelsState,
            numberOfBars: 6
        });
        expect(result).toBe(null);
    });
});
