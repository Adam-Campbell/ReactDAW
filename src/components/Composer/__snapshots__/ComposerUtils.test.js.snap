// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createGridLinesArray correctly creates and returns the array 1`] = `
Array [
  Array [
    0,
    40,
    0,
    320,
  ],
  Array [
    48,
    40,
    48,
    320,
  ],
  Array [
    96,
    40,
    96,
    320,
  ],
  Array [
    144,
    40,
    144,
    320,
  ],
  Array [
    192,
    40,
    192,
    320,
  ],
  Array [
    240,
    40,
    240,
    320,
  ],
  Array [
    288,
    40,
    288,
    320,
  ],
  Array [
    336,
    40,
    336,
    320,
  ],
  Array [
    384,
    40,
    384,
    320,
  ],
  Array [
    432,
    40,
    432,
    320,
  ],
  Array [
    480,
    40,
    480,
    320,
  ],
  Array [
    528,
    40,
    528,
    320,
  ],
  Array [
    576,
    40,
    576,
    320,
  ],
  Array [
    624,
    40,
    624,
    320,
  ],
  Array [
    672,
    40,
    672,
    320,
  ],
  Array [
    720,
    40,
    720,
    320,
  ],
  Array [
    768,
    40,
    768,
    320,
  ],
  Array [
    816,
    40,
    816,
    320,
  ],
  Array [
    864,
    40,
    864,
    320,
  ],
  Array [
    912,
    40,
    912,
    320,
  ],
  Array [
    960,
    40,
    960,
    320,
  ],
  Array [
    1008,
    40,
    1008,
    320,
  ],
  Array [
    1056,
    40,
    1056,
    320,
  ],
  Array [
    1104,
    40,
    1104,
    320,
  ],
  Array [
    1152,
    40,
    1152,
    320,
  ],
  Array [
    1200,
    40,
    1200,
    320,
  ],
  Array [
    1248,
    40,
    1248,
    320,
  ],
  Array [
    1296,
    40,
    1296,
    320,
  ],
  Array [
    1344,
    40,
    1344,
    320,
  ],
  Array [
    1392,
    40,
    1392,
    320,
  ],
  Array [
    1440,
    40,
    1440,
    320,
  ],
  Array [
    1488,
    40,
    1488,
    320,
  ],
  Array [
    1536,
    40,
    1536,
    320,
  ],
  Array [
    1584,
    40,
    1584,
    320,
  ],
  Array [
    1632,
    40,
    1632,
    320,
  ],
  Array [
    1680,
    40,
    1680,
    320,
  ],
  Array [
    1728,
    40,
    1728,
    320,
  ],
  Array [
    1776,
    40,
    1776,
    320,
  ],
  Array [
    1824,
    40,
    1824,
    320,
  ],
  Array [
    1872,
    40,
    1872,
    320,
  ],
  Array [
    1920,
    40,
    1920,
    320,
  ],
  Array [
    1968,
    40,
    1968,
    320,
  ],
  Array [
    2016,
    40,
    2016,
    320,
  ],
  Array [
    2064,
    40,
    2064,
    320,
  ],
  Array [
    2112,
    40,
    2112,
    320,
  ],
  Array [
    2160,
    40,
    2160,
    320,
  ],
  Array [
    2208,
    40,
    2208,
    320,
  ],
  Array [
    2256,
    40,
    2256,
    320,
  ],
  Array [
    2304,
    40,
    2304,
    320,
  ],
  Array [
    2352,
    40,
    2352,
    320,
  ],
  Array [
    2400,
    40,
    2400,
    320,
  ],
  Array [
    2448,
    40,
    2448,
    320,
  ],
  Array [
    2496,
    40,
    2496,
    320,
  ],
  Array [
    2544,
    40,
    2544,
    320,
  ],
  Array [
    2592,
    40,
    2592,
    320,
  ],
  Array [
    2640,
    40,
    2640,
    320,
  ],
  Array [
    2688,
    40,
    2688,
    320,
  ],
  Array [
    2736,
    40,
    2736,
    320,
  ],
  Array [
    2784,
    40,
    2784,
    320,
  ],
  Array [
    2832,
    40,
    2832,
    320,
  ],
  Array [
    2880,
    40,
    2880,
    320,
  ],
  Array [
    2928,
    40,
    2928,
    320,
  ],
  Array [
    2976,
    40,
    2976,
    320,
  ],
  Array [
    3024,
    40,
    3024,
    320,
  ],
  Array [
    3072,
    40,
    3072,
    320,
  ],
  Array [
    3120,
    40,
    3120,
    320,
  ],
  Array [
    3168,
    40,
    3168,
    320,
  ],
  Array [
    3216,
    40,
    3216,
    320,
  ],
  Array [
    3264,
    40,
    3264,
    320,
  ],
  Array [
    3312,
    40,
    3312,
    320,
  ],
  Array [
    3360,
    40,
    3360,
    320,
  ],
  Array [
    3408,
    40,
    3408,
    320,
  ],
  Array [
    3456,
    40,
    3456,
    320,
  ],
  Array [
    3504,
    40,
    3504,
    320,
  ],
  Array [
    3552,
    40,
    3552,
    320,
  ],
  Array [
    3600,
    40,
    3600,
    320,
  ],
  Array [
    3648,
    40,
    3648,
    320,
  ],
  Array [
    3696,
    40,
    3696,
    320,
  ],
  Array [
    3744,
    40,
    3744,
    320,
  ],
  Array [
    3792,
    40,
    3792,
    320,
  ],
  Array [
    3840,
    40,
    3840,
    320,
  ],
  Array [
    3888,
    40,
    3888,
    320,
  ],
  Array [
    3936,
    40,
    3936,
    320,
  ],
  Array [
    3984,
    40,
    3984,
    320,
  ],
  Array [
    4032,
    40,
    4032,
    320,
  ],
  Array [
    4080,
    40,
    4080,
    320,
  ],
  Array [
    4128,
    40,
    4128,
    320,
  ],
  Array [
    4176,
    40,
    4176,
    320,
  ],
  Array [
    4224,
    40,
    4224,
    320,
  ],
  Array [
    4272,
    40,
    4272,
    320,
  ],
  Array [
    4320,
    40,
    4320,
    320,
  ],
  Array [
    4368,
    40,
    4368,
    320,
  ],
  Array [
    4416,
    40,
    4416,
    320,
  ],
  Array [
    4464,
    40,
    4464,
    320,
  ],
  Array [
    4512,
    40,
    4512,
    320,
  ],
  Array [
    4560,
    40,
    4560,
    320,
  ],
  Array [
    4608,
    40,
    4608,
    320,
  ],
  Array [
    4656,
    40,
    4656,
    320,
  ],
  Array [
    4704,
    40,
    4704,
    320,
  ],
  Array [
    4752,
    40,
    4752,
    320,
  ],
  Array [
    4800,
    40,
    4800,
    320,
  ],
  Array [
    4848,
    40,
    4848,
    320,
  ],
  Array [
    4896,
    40,
    4896,
    320,
  ],
  Array [
    4944,
    40,
    4944,
    320,
  ],
  Array [
    4992,
    40,
    4992,
    320,
  ],
  Array [
    5040,
    40,
    5040,
    320,
  ],
  Array [
    5088,
    40,
    5088,
    320,
  ],
  Array [
    5136,
    40,
    5136,
    320,
  ],
  Array [
    5184,
    40,
    5184,
    320,
  ],
  Array [
    5232,
    40,
    5232,
    320,
  ],
  Array [
    5280,
    40,
    5280,
    320,
  ],
  Array [
    5328,
    40,
    5328,
    320,
  ],
  Array [
    5376,
    40,
    5376,
    320,
  ],
  Array [
    5424,
    40,
    5424,
    320,
  ],
  Array [
    5472,
    40,
    5472,
    320,
  ],
  Array [
    5520,
    40,
    5520,
    320,
  ],
  Array [
    5568,
    40,
    5568,
    320,
  ],
  Array [
    5616,
    40,
    5616,
    320,
  ],
  Array [
    5664,
    40,
    5664,
    320,
  ],
  Array [
    5712,
    40,
    5712,
    320,
  ],
  Array [
    5760,
    40,
    5760,
    320,
  ],
  Array [
    5808,
    40,
    5808,
    320,
  ],
  Array [
    5856,
    40,
    5856,
    320,
  ],
  Array [
    5904,
    40,
    5904,
    320,
  ],
  Array [
    5952,
    40,
    5952,
    320,
  ],
  Array [
    6000,
    40,
    6000,
    320,
  ],
  Array [
    6048,
    40,
    6048,
    320,
  ],
  Array [
    6096,
    40,
    6096,
    320,
  ],
  Array [
    6144,
    40,
    6144,
    320,
  ],
  Array [
    6192,
    40,
    6192,
    320,
  ],
  Array [
    6240,
    40,
    6240,
    320,
  ],
  Array [
    6288,
    40,
    6288,
    320,
  ],
  Array [
    6336,
    40,
    6336,
    320,
  ],
  Array [
    6384,
    40,
    6384,
    320,
  ],
  Array [
    6432,
    40,
    6432,
    320,
  ],
  Array [
    6480,
    40,
    6480,
    320,
  ],
  Array [
    6528,
    40,
    6528,
    320,
  ],
  Array [
    6576,
    40,
    6576,
    320,
  ],
  Array [
    6624,
    40,
    6624,
    320,
  ],
  Array [
    6672,
    40,
    6672,
    320,
  ],
  Array [
    6720,
    40,
    6720,
    320,
  ],
  Array [
    6768,
    40,
    6768,
    320,
  ],
  Array [
    6816,
    40,
    6816,
    320,
  ],
  Array [
    6864,
    40,
    6864,
    320,
  ],
  Array [
    6912,
    40,
    6912,
    320,
  ],
  Array [
    6960,
    40,
    6960,
    320,
  ],
  Array [
    7008,
    40,
    7008,
    320,
  ],
  Array [
    7056,
    40,
    7056,
    320,
  ],
  Array [
    7104,
    40,
    7104,
    320,
  ],
  Array [
    7152,
    40,
    7152,
    320,
  ],
  Array [
    7200,
    40,
    7200,
    320,
  ],
  Array [
    7248,
    40,
    7248,
    320,
  ],
  Array [
    7296,
    40,
    7296,
    320,
  ],
  Array [
    7344,
    40,
    7344,
    320,
  ],
  Array [
    7392,
    40,
    7392,
    320,
  ],
  Array [
    7440,
    40,
    7440,
    320,
  ],
  Array [
    7488,
    40,
    7488,
    320,
  ],
  Array [
    7536,
    40,
    7536,
    320,
  ],
  Array [
    7584,
    40,
    7584,
    320,
  ],
  Array [
    7632,
    40,
    7632,
    320,
  ],
  Array [
    7680,
    40,
    7680,
    320,
  ],
  Array [
    7728,
    40,
    7728,
    320,
  ],
  Array [
    7776,
    40,
    7776,
    320,
  ],
  Array [
    7824,
    40,
    7824,
    320,
  ],
  Array [
    7872,
    40,
    7872,
    320,
  ],
  Array [
    7920,
    40,
    7920,
    320,
  ],
  Array [
    7968,
    40,
    7968,
    320,
  ],
  Array [
    8016,
    40,
    8016,
    320,
  ],
  Array [
    8064,
    40,
    8064,
    320,
  ],
  Array [
    8112,
    40,
    8112,
    320,
  ],
  Array [
    8160,
    40,
    8160,
    320,
  ],
  Array [
    8208,
    40,
    8208,
    320,
  ],
  Array [
    8256,
    40,
    8256,
    320,
  ],
  Array [
    8304,
    40,
    8304,
    320,
  ],
  Array [
    8352,
    40,
    8352,
    320,
  ],
  Array [
    8400,
    40,
    8400,
    320,
  ],
  Array [
    8448,
    40,
    8448,
    320,
  ],
  Array [
    8496,
    40,
    8496,
    320,
  ],
  Array [
    8544,
    40,
    8544,
    320,
  ],
  Array [
    8592,
    40,
    8592,
    320,
  ],
  Array [
    8640,
    40,
    8640,
    320,
  ],
  Array [
    8688,
    40,
    8688,
    320,
  ],
  Array [
    8736,
    40,
    8736,
    320,
  ],
  Array [
    8784,
    40,
    8784,
    320,
  ],
  Array [
    8832,
    40,
    8832,
    320,
  ],
  Array [
    8880,
    40,
    8880,
    320,
  ],
  Array [
    8928,
    40,
    8928,
    320,
  ],
  Array [
    8976,
    40,
    8976,
    320,
  ],
  Array [
    9024,
    40,
    9024,
    320,
  ],
  Array [
    9072,
    40,
    9072,
    320,
  ],
  Array [
    9120,
    40,
    9120,
    320,
  ],
  Array [
    9168,
    40,
    9168,
    320,
  ],
  Array [
    9216,
    40,
    9216,
    320,
  ],
  Array [
    9264,
    40,
    9264,
    320,
  ],
  Array [
    9312,
    40,
    9312,
    320,
  ],
  Array [
    9360,
    40,
    9360,
    320,
  ],
  Array [
    9408,
    40,
    9408,
    320,
  ],
  Array [
    9456,
    40,
    9456,
    320,
  ],
  Array [
    9504,
    40,
    9504,
    320,
  ],
  Array [
    9552,
    40,
    9552,
    320,
  ],
  Array [
    9600,
    40,
    9600,
    320,
  ],
  Array [
    0,
    40,
    9600,
    40,
  ],
  Array [
    0,
    110,
    9600,
    110,
  ],
  Array [
    0,
    180,
    9600,
    180,
  ],
  Array [
    0,
    250,
    9600,
    250,
  ],
  Array [
    0,
    320,
    9600,
    320,
  ],
]
`;

exports[`createGridLinesArray correctly creates and returns the array 2`] = `
Array [
  Array [
    0,
    40,
    0,
    530,
  ],
  Array [
    48,
    40,
    48,
    530,
  ],
  Array [
    96,
    40,
    96,
    530,
  ],
  Array [
    144,
    40,
    144,
    530,
  ],
  Array [
    192,
    40,
    192,
    530,
  ],
  Array [
    240,
    40,
    240,
    530,
  ],
  Array [
    288,
    40,
    288,
    530,
  ],
  Array [
    336,
    40,
    336,
    530,
  ],
  Array [
    384,
    40,
    384,
    530,
  ],
  Array [
    432,
    40,
    432,
    530,
  ],
  Array [
    480,
    40,
    480,
    530,
  ],
  Array [
    528,
    40,
    528,
    530,
  ],
  Array [
    576,
    40,
    576,
    530,
  ],
  Array [
    624,
    40,
    624,
    530,
  ],
  Array [
    672,
    40,
    672,
    530,
  ],
  Array [
    720,
    40,
    720,
    530,
  ],
  Array [
    768,
    40,
    768,
    530,
  ],
  Array [
    816,
    40,
    816,
    530,
  ],
  Array [
    864,
    40,
    864,
    530,
  ],
  Array [
    912,
    40,
    912,
    530,
  ],
  Array [
    960,
    40,
    960,
    530,
  ],
  Array [
    1008,
    40,
    1008,
    530,
  ],
  Array [
    1056,
    40,
    1056,
    530,
  ],
  Array [
    1104,
    40,
    1104,
    530,
  ],
  Array [
    1152,
    40,
    1152,
    530,
  ],
  Array [
    1200,
    40,
    1200,
    530,
  ],
  Array [
    1248,
    40,
    1248,
    530,
  ],
  Array [
    1296,
    40,
    1296,
    530,
  ],
  Array [
    1344,
    40,
    1344,
    530,
  ],
  Array [
    1392,
    40,
    1392,
    530,
  ],
  Array [
    1440,
    40,
    1440,
    530,
  ],
  Array [
    1488,
    40,
    1488,
    530,
  ],
  Array [
    1536,
    40,
    1536,
    530,
  ],
  Array [
    1584,
    40,
    1584,
    530,
  ],
  Array [
    1632,
    40,
    1632,
    530,
  ],
  Array [
    1680,
    40,
    1680,
    530,
  ],
  Array [
    1728,
    40,
    1728,
    530,
  ],
  Array [
    1776,
    40,
    1776,
    530,
  ],
  Array [
    1824,
    40,
    1824,
    530,
  ],
  Array [
    1872,
    40,
    1872,
    530,
  ],
  Array [
    1920,
    40,
    1920,
    530,
  ],
  Array [
    1968,
    40,
    1968,
    530,
  ],
  Array [
    2016,
    40,
    2016,
    530,
  ],
  Array [
    2064,
    40,
    2064,
    530,
  ],
  Array [
    2112,
    40,
    2112,
    530,
  ],
  Array [
    2160,
    40,
    2160,
    530,
  ],
  Array [
    2208,
    40,
    2208,
    530,
  ],
  Array [
    2256,
    40,
    2256,
    530,
  ],
  Array [
    2304,
    40,
    2304,
    530,
  ],
  Array [
    2352,
    40,
    2352,
    530,
  ],
  Array [
    2400,
    40,
    2400,
    530,
  ],
  Array [
    2448,
    40,
    2448,
    530,
  ],
  Array [
    2496,
    40,
    2496,
    530,
  ],
  Array [
    2544,
    40,
    2544,
    530,
  ],
  Array [
    2592,
    40,
    2592,
    530,
  ],
  Array [
    2640,
    40,
    2640,
    530,
  ],
  Array [
    2688,
    40,
    2688,
    530,
  ],
  Array [
    2736,
    40,
    2736,
    530,
  ],
  Array [
    2784,
    40,
    2784,
    530,
  ],
  Array [
    2832,
    40,
    2832,
    530,
  ],
  Array [
    2880,
    40,
    2880,
    530,
  ],
  Array [
    2928,
    40,
    2928,
    530,
  ],
  Array [
    2976,
    40,
    2976,
    530,
  ],
  Array [
    3024,
    40,
    3024,
    530,
  ],
  Array [
    3072,
    40,
    3072,
    530,
  ],
  Array [
    3120,
    40,
    3120,
    530,
  ],
  Array [
    3168,
    40,
    3168,
    530,
  ],
  Array [
    3216,
    40,
    3216,
    530,
  ],
  Array [
    3264,
    40,
    3264,
    530,
  ],
  Array [
    3312,
    40,
    3312,
    530,
  ],
  Array [
    3360,
    40,
    3360,
    530,
  ],
  Array [
    3408,
    40,
    3408,
    530,
  ],
  Array [
    3456,
    40,
    3456,
    530,
  ],
  Array [
    3504,
    40,
    3504,
    530,
  ],
  Array [
    3552,
    40,
    3552,
    530,
  ],
  Array [
    3600,
    40,
    3600,
    530,
  ],
  Array [
    3648,
    40,
    3648,
    530,
  ],
  Array [
    3696,
    40,
    3696,
    530,
  ],
  Array [
    3744,
    40,
    3744,
    530,
  ],
  Array [
    3792,
    40,
    3792,
    530,
  ],
  Array [
    3840,
    40,
    3840,
    530,
  ],
  Array [
    3888,
    40,
    3888,
    530,
  ],
  Array [
    3936,
    40,
    3936,
    530,
  ],
  Array [
    3984,
    40,
    3984,
    530,
  ],
  Array [
    4032,
    40,
    4032,
    530,
  ],
  Array [
    4080,
    40,
    4080,
    530,
  ],
  Array [
    4128,
    40,
    4128,
    530,
  ],
  Array [
    4176,
    40,
    4176,
    530,
  ],
  Array [
    4224,
    40,
    4224,
    530,
  ],
  Array [
    4272,
    40,
    4272,
    530,
  ],
  Array [
    4320,
    40,
    4320,
    530,
  ],
  Array [
    4368,
    40,
    4368,
    530,
  ],
  Array [
    4416,
    40,
    4416,
    530,
  ],
  Array [
    4464,
    40,
    4464,
    530,
  ],
  Array [
    4512,
    40,
    4512,
    530,
  ],
  Array [
    4560,
    40,
    4560,
    530,
  ],
  Array [
    4608,
    40,
    4608,
    530,
  ],
  Array [
    4656,
    40,
    4656,
    530,
  ],
  Array [
    4704,
    40,
    4704,
    530,
  ],
  Array [
    4752,
    40,
    4752,
    530,
  ],
  Array [
    4800,
    40,
    4800,
    530,
  ],
  Array [
    4848,
    40,
    4848,
    530,
  ],
  Array [
    4896,
    40,
    4896,
    530,
  ],
  Array [
    4944,
    40,
    4944,
    530,
  ],
  Array [
    4992,
    40,
    4992,
    530,
  ],
  Array [
    5040,
    40,
    5040,
    530,
  ],
  Array [
    5088,
    40,
    5088,
    530,
  ],
  Array [
    5136,
    40,
    5136,
    530,
  ],
  Array [
    5184,
    40,
    5184,
    530,
  ],
  Array [
    5232,
    40,
    5232,
    530,
  ],
  Array [
    5280,
    40,
    5280,
    530,
  ],
  Array [
    5328,
    40,
    5328,
    530,
  ],
  Array [
    5376,
    40,
    5376,
    530,
  ],
  Array [
    5424,
    40,
    5424,
    530,
  ],
  Array [
    5472,
    40,
    5472,
    530,
  ],
  Array [
    5520,
    40,
    5520,
    530,
  ],
  Array [
    5568,
    40,
    5568,
    530,
  ],
  Array [
    5616,
    40,
    5616,
    530,
  ],
  Array [
    5664,
    40,
    5664,
    530,
  ],
  Array [
    5712,
    40,
    5712,
    530,
  ],
  Array [
    5760,
    40,
    5760,
    530,
  ],
  Array [
    5808,
    40,
    5808,
    530,
  ],
  Array [
    5856,
    40,
    5856,
    530,
  ],
  Array [
    5904,
    40,
    5904,
    530,
  ],
  Array [
    5952,
    40,
    5952,
    530,
  ],
  Array [
    6000,
    40,
    6000,
    530,
  ],
  Array [
    6048,
    40,
    6048,
    530,
  ],
  Array [
    6096,
    40,
    6096,
    530,
  ],
  Array [
    6144,
    40,
    6144,
    530,
  ],
  Array [
    6192,
    40,
    6192,
    530,
  ],
  Array [
    6240,
    40,
    6240,
    530,
  ],
  Array [
    6288,
    40,
    6288,
    530,
  ],
  Array [
    6336,
    40,
    6336,
    530,
  ],
  Array [
    6384,
    40,
    6384,
    530,
  ],
  Array [
    6432,
    40,
    6432,
    530,
  ],
  Array [
    6480,
    40,
    6480,
    530,
  ],
  Array [
    6528,
    40,
    6528,
    530,
  ],
  Array [
    6576,
    40,
    6576,
    530,
  ],
  Array [
    6624,
    40,
    6624,
    530,
  ],
  Array [
    6672,
    40,
    6672,
    530,
  ],
  Array [
    6720,
    40,
    6720,
    530,
  ],
  Array [
    6768,
    40,
    6768,
    530,
  ],
  Array [
    6816,
    40,
    6816,
    530,
  ],
  Array [
    6864,
    40,
    6864,
    530,
  ],
  Array [
    6912,
    40,
    6912,
    530,
  ],
  Array [
    6960,
    40,
    6960,
    530,
  ],
  Array [
    7008,
    40,
    7008,
    530,
  ],
  Array [
    7056,
    40,
    7056,
    530,
  ],
  Array [
    7104,
    40,
    7104,
    530,
  ],
  Array [
    7152,
    40,
    7152,
    530,
  ],
  Array [
    7200,
    40,
    7200,
    530,
  ],
  Array [
    7248,
    40,
    7248,
    530,
  ],
  Array [
    7296,
    40,
    7296,
    530,
  ],
  Array [
    7344,
    40,
    7344,
    530,
  ],
  Array [
    7392,
    40,
    7392,
    530,
  ],
  Array [
    7440,
    40,
    7440,
    530,
  ],
  Array [
    7488,
    40,
    7488,
    530,
  ],
  Array [
    7536,
    40,
    7536,
    530,
  ],
  Array [
    7584,
    40,
    7584,
    530,
  ],
  Array [
    7632,
    40,
    7632,
    530,
  ],
  Array [
    7680,
    40,
    7680,
    530,
  ],
  Array [
    7728,
    40,
    7728,
    530,
  ],
  Array [
    7776,
    40,
    7776,
    530,
  ],
  Array [
    7824,
    40,
    7824,
    530,
  ],
  Array [
    7872,
    40,
    7872,
    530,
  ],
  Array [
    7920,
    40,
    7920,
    530,
  ],
  Array [
    7968,
    40,
    7968,
    530,
  ],
  Array [
    8016,
    40,
    8016,
    530,
  ],
  Array [
    8064,
    40,
    8064,
    530,
  ],
  Array [
    8112,
    40,
    8112,
    530,
  ],
  Array [
    8160,
    40,
    8160,
    530,
  ],
  Array [
    8208,
    40,
    8208,
    530,
  ],
  Array [
    8256,
    40,
    8256,
    530,
  ],
  Array [
    8304,
    40,
    8304,
    530,
  ],
  Array [
    8352,
    40,
    8352,
    530,
  ],
  Array [
    8400,
    40,
    8400,
    530,
  ],
  Array [
    8448,
    40,
    8448,
    530,
  ],
  Array [
    8496,
    40,
    8496,
    530,
  ],
  Array [
    8544,
    40,
    8544,
    530,
  ],
  Array [
    8592,
    40,
    8592,
    530,
  ],
  Array [
    8640,
    40,
    8640,
    530,
  ],
  Array [
    8688,
    40,
    8688,
    530,
  ],
  Array [
    8736,
    40,
    8736,
    530,
  ],
  Array [
    8784,
    40,
    8784,
    530,
  ],
  Array [
    8832,
    40,
    8832,
    530,
  ],
  Array [
    8880,
    40,
    8880,
    530,
  ],
  Array [
    8928,
    40,
    8928,
    530,
  ],
  Array [
    8976,
    40,
    8976,
    530,
  ],
  Array [
    9024,
    40,
    9024,
    530,
  ],
  Array [
    9072,
    40,
    9072,
    530,
  ],
  Array [
    9120,
    40,
    9120,
    530,
  ],
  Array [
    9168,
    40,
    9168,
    530,
  ],
  Array [
    9216,
    40,
    9216,
    530,
  ],
  Array [
    9264,
    40,
    9264,
    530,
  ],
  Array [
    9312,
    40,
    9312,
    530,
  ],
  Array [
    9360,
    40,
    9360,
    530,
  ],
  Array [
    9408,
    40,
    9408,
    530,
  ],
  Array [
    9456,
    40,
    9456,
    530,
  ],
  Array [
    9504,
    40,
    9504,
    530,
  ],
  Array [
    9552,
    40,
    9552,
    530,
  ],
  Array [
    9600,
    40,
    9600,
    530,
  ],
  Array [
    0,
    40,
    12000,
    40,
  ],
  Array [
    0,
    110,
    12000,
    110,
  ],
  Array [
    0,
    180,
    12000,
    180,
  ],
  Array [
    0,
    250,
    12000,
    250,
  ],
  Array [
    0,
    320,
    12000,
    320,
  ],
  Array [
    0,
    390,
    12000,
    390,
  ],
  Array [
    0,
    460,
    12000,
    460,
  ],
  Array [
    0,
    530,
    12000,
    530,
  ],
]
`;

exports[`createSectionRectsArray correctly creates and returns the array 1`] = `
Array [
  Object {
    "color": "#00bff3",
    "height": 70,
    "sectionId": "3873914619840634",
    "width": 48,
    "x": 192,
    "y": 40,
  },
  Object {
    "color": "#00bff3",
    "height": 70,
    "sectionId": "5586134390473357",
    "width": 48,
    "x": 288,
    "y": 40,
  },
  Object {
    "color": "#00bff3",
    "height": 70,
    "sectionId": "4928704574589138",
    "width": 96,
    "x": 384,
    "y": 40,
  },
  Object {
    "color": "#f4ab54",
    "height": 70,
    "sectionId": "5208036813028087",
    "width": 192,
    "x": 0,
    "y": 110,
  },
  Object {
    "color": "#f4ab54",
    "height": 70,
    "sectionId": "6955590810822236",
    "width": 192,
    "x": 192,
    "y": 110,
  },
  Object {
    "color": "#f4ab54",
    "height": 70,
    "sectionId": "9199138957540981",
    "width": 96,
    "x": 480,
    "y": 110,
  },
]
`;
